xapian-core/api/positioniterator.cc

```diff
@@ -76,10 +78,19 @@ PositionIterator::operator=(const PositionIterator & o)
     RETURN(*this);
 }
 
-PositionIterator::PositionIterator(PositionIterator &&) = default;
+PositionIterator::PositionIterator(PositionIterator && o)
+    : internal(o.internal)
+{
```

Comment:
Should I add LOG same as copy ctor & assignment op(except object pointer)?

Reply:
I don't think logging is all that useful for these.  The compiler is allowed (and sometimes required) to eliminate calls to them, so logging wouldn't correspond to the source exactly anyway.

The only reason I can think that logging might be interesting is to try to find where copies aren't eliminated to try to address that.  But by inlining we open things up for the compiler to optimise in more ways, and without the programmer having to study logs.  The savings aren't likely to be very large anyway, as copying just copies a pointer and adjusts reference counts.

Overall having these inline in the API headers seems a better trade-off than putting them into the library so they can have debug logging.